src/components/Api.js

```diff
@@ -0,0 +1,155 @@
+export default class Api {
+  constructor(options) {
+    this._url = options.baseUrl;
+    this._headers = options.headers;
+    // this._test =
+  }
+
+  getUserInfo() {
+    return fetch(`${this._url}/users/me`, {
+      headers: this._headers,
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      }
+      return Promise.reject(`Error: ${res.status}`);
+    });
+  }
+
+  setUserInfo(values, submitButton) {
+    submitButton.textContent = "Saving ...";
```

Comment:
`    submitButton.textContent = "Saving ...";
`

Please, delete submit buttons handling from `Api`. It's not a responsibility of this class. It handles only requests and checking responses.